File: frontend/src/types/response/email.ts
import { z } from "zod";

export const GetEmailLogSchema = z.object({
    id: z.string().uuid(),
    to: z.string().email(),
    subject: z.string(),
    body: z.string(),
    status: z.enum(["pending", "sent", "failed"]),
    created_at: z.string(),
});

export type GetEmailLog = z.infer<typeof GetEmailLogSchema>;

File: backend/model/response/email.go
package response

import "github.com/google/uuid"

// Email represents an email record
type GetEmailLog struct {
	ID        uuid.UUID `json:"id"`
	To        string    `json:"to"`
	Subject   string    `json:"subject"`
	Body      string    `json:"body"`
	Status    string    `json:"status"` // "pending", "sent", "failed"
	CreatedAt string    `json:"created_at"`
}
